### ISSUE TITLE
an empty target for easier scripting

### ISSUE THREAD

**Issue by gasche** [ocaml/dune#4161]
## Desired Behavior

In a CI script I want to let users specify their testing targets (`"@runtest"` by default, but maybe it's just `"@tests/fast/runtest @tests/somewhat-fast/runtest"` for example), with the convention that if they give an empty string, then tests will not run at all.

Currently I write this in the following way:

```
if [ -n "$TEST_TARGETS" ]; then dune build $TEST_TARGETS; fi
```

But I would like to avoid the conditional by saying Dune: if I don't give you any target, please don't do anything. So something like `dune build --dont-do-anything-by-default $TEST_TARGETS`.

This is almost possible with the option `dune build --default-target="..." $TEST_TARGETS`, except that I have not found a way to specify a default target that *does nothing*. If I use a non-existing alias such as `@empty` there, I get an error and the build fails.

Feature request: provide an explicit target (`@none` or `@empty` for example) that successfully does nothing, so that I can use it as default target for my use-case.

**Comment by jeremiedimino** [comment]
Not necessarily against the idea, but what is the problem with the conditional?

**Comment by gasche** [comment]
In CI scripts (or cram tests) we show the command that is performed to have some observable / self-explanatory output, the readability of the command in the logs is important. For me there is a clear advantage to:

```
dune build --default--target=@empty $DUNE_TEST_TARGETS
```
over
```
if [ -n "$DUNE_TEST_TARGETS" ]; then dune build $DUNE_TEST_TARGETS; fi
```

It's not enough of an argument to ask for a lot of work on the Dune side, or a specific feature to hardcode this use-case, but I think the feature request is general enough, probably easy to implement, and may be useful in other unforeseen scenarios (it makes the target language more regular).

Now that I think of it: in fact we could even ask users to use `@empty` instead of the empty string if they want to skip this step, and it would give even simpler code.

**Comment by jeremiedimino** [comment]
Makes sense. Another name suggestion: `@nothing`. It reads quite well: `dune build @nothing`.

So we can change dune right now so that it doesn't complain if `@nothing` is empty. It's a one line change [here](https://github.com/ocaml/dune/blob/34d654a799263bd830d59352b53cf76d8f79410b/src/dune_engine/alias.ml#L188), you need to add `let nothing = make_standard (Name.of_string "nothing")`. Happy to accept a PR for that.

Then to be complete, we should probably forbid people from attaching things to the `nothing` alias. However, this is technically a breaking change so such a check should only be introduced with version 3.0 of the language.

**Comment by rgrinberg** [comment]
@snowleopard what do you think about this idea? Should we bless this special alias in the engine?

**Comment by snowleopard** [comment]
Adding an `@empty` alias sounds great to me. We have internally and it's quite useful.

**Comment by snowleopard** [comment]
(The name `@nothing` sounds good too but I think I prefer `empty` because I tend to think of aliases as sets.)

**Comment by rgrinberg** [comment]
What happens when a user tries to attach a dependency to this `empty` alias? For example:

```
(alias
 (name empty)
 (deps foo))
```

**Comment by rgrinberg** [comment]
@snowleopard gentle ping on the question

**Comment by snowleopard** [comment]
Good question! I just tested it internally and this doesn't produce an error, which I think is a mistake. We shouldn't allow to extend the `empty` alias.
